### PR TITLE
fix(SR-2): vulnerability fixes on TFM API

### PR DESCRIPTION
**Pull request by ChristianMcc**
## Introduction
Fix for SR-2. TFM API had no authentication for API calls

## Resolution
* Added authRouter with `jwt` to TFM-API routes
* Added openRouter with `x-api-header` key for deal submit calls and create user call
* Updated mockDataLoader to use token
* Updated unit tests
* Updated e2e tests to login to tfm and use token to submit deal
* Changed mocks companies house number from `08547313` to `06771815` as was not recognised by dev partyDB
* Updated mocked deals.
* Refactored `try-catch` blocks.
* Allowed `404` as valid response for `axios` call.

## Miscellaneous
* Updated dependencies.

### CODE REVIEW

File: trade-finance-manager-api/api-tests/v1/feedback/feedback.api-test.js [hunk 1]
```diff
@@ -18,13 +19,15 @@ describe('/feedback', () => {
 
   describe('POST /v1/feedback', () => {
     it('it successfully sends feedback form', async () => {
-      const { status } = await api.post(feedbackFormBody).to('/v1/feedback');
+      const user = await testUserCache.initialise(app);
+      const { status } = await as(user).post(feedbackFormBody).to('/v1/feedback');
```
Comment (ChristianMcc) [inline trade-finance-manager-api/api-tests/v1/feedback/feedback.api-test.js:23]: These could be reverted to old tests to check they work not logged in now

File: trade-finance-manager-ui/server/controllers/feedback/index.js [hunk 2]
```diff
@@ -22,11 +22,13 @@ const postFeedback = async (req, res) => {
       userDetails.username = username;
       userDetails.email = email;
     }
+    // TODO: Protecting this behind JWT needing to be logged in nullifies the req.session.user above
```
Comment (ChristianMcc) [inline trade-finance-manager-ui/server/controllers/feedback/index.js:25]: remove this as token unnecessary now

**Review by ChristianMcc — COMMENTED** [review]
Haven't found much - I can just action my own comments just putting them here for visibility. Main thing is just fixing the feedback controller to be API key based instead of JWT again.

Might look at making API keys in API tests more specific as well given I've got some time

**Review by Zainzzkk — APPROVED** [review]

File: utils/mock-data-loader/temporary-token-handler-tfm.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const axios = require('axios');
+require('dotenv').config();
+
+const urlRoot = process.env.TFM_API;
```
Comment (IgnatG) [inline utils/mock-data-loader/temporary-token-handler-tfm.js:4]: this line and the one below can be merged into 1

**Review by IgnatG — DISMISSED** [review]

File: utils/mock-data-loader/temporary-token-handler-tfm.js (hunk 3 above)
Comment (abhi-markan) [reply utils/mock-data-loader/temporary-token-handler-tfm.js:4]: Have updated.

File: utils/mock-data-loader/temporary-token-handler-tfm.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const { TFM_API_KEY } = process.env;
+
+module.exports = async (user) => {
+  console.info(`Creating temp user "${user.username}"`);
+  await axios({
+    method: 'post',
+    url: `${urlRoot}/v1/user`,
+    headers: {
+      'Content-Type': 'application/json',
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply utils/mock-data-loader/temporary-token-handler-tfm.js:13]: Default value can vary from `html` to `octet`, always best to explicitly declare I think.

**Review by Zainzzkk — DISMISSED** [review]

File: utils/mock-data-loader/temporary-token-handler-tfm.js [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-const urlRoot = process.env.TFM_API;
-const { TFM_API_KEY } = process.env;
+const { TFM_API, TFM_API_KEY } = process.env;
 
 module.exports = async (user) => {
   console.info(`Creating temp user "${user.username}"`);
   await axios({
     method: 'post',
-    url: `${urlRoot}/v1/user`,
+    TFM_API: `${TFM_API}/v1/user`,
```
Comment (IgnatG) [inline utils/mock-data-loader/temporary-token-handler-tfm.js:10]: this needs to be `url`

**Pull request merged**